src/test/kotlin/org/jmailen/gradle/kotlinter/functional/ReportersTest.kt

```diff
@@ -0,0 +1,216 @@
+package org.jmailen.gradle.kotlinter.functional
+
+import org.gradle.testkit.runner.TaskOutcome
+import org.jmailen.gradle.kotlinter.functional.utils.kotlinClass
+import org.jmailen.gradle.kotlinter.functional.utils.resolve
+import org.jmailen.gradle.kotlinter.functional.utils.settingsFile
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.condition.DisabledOnOs
+import org.junit.jupiter.api.condition.OS
+import java.io.File
+
+class ReportersTest : WithGradleTest.Kotlin() {
+
+    lateinit var projectRoot: File
+
+    @BeforeEach
+    fun setUp() {
+        projectRoot = testProjectDir.apply {
+            resolve("settings.gradle") { writeText(settingsFile) }
+            resolve("build.gradle") {
+                // language=groovy
+                val buildScript =
+                    """
+                    plugins {
+                        id 'kotlin'
+                        id 'org.jmailen.kotlinter'
+                    }
+                    
+                    repositories {
+                        mavenCentral()
+                    }
+                
+                    """.trimIndent()
+                writeText(buildScript)
+            }
+        }
+        projectRoot.resolve("src/main/kotlin/CustomClass.kt") {
+            writeText(kotlinClass("CustomClass"))
+        }
+    }
+
+    @Test
+    @DisabledOnOs(OS.WINDOWS, disabledReason = "Report file content differs on Windows due to different path separator")
+    fun `supports all types of reporters`() {
+        projectRoot.resolve("build.gradle") {
+            // language=groovy
+            val buildScript =
+                """
+
+                kotlinter {
+                    reporters = [
+                        'checkstyle',
+                        'html',
+                        'json',
+                        'plain',
+                        'sarif',
+                    ] 
+                }
+
+                """
+            appendText(buildScript)
+        }
+
+        build("lintKotlin").apply {
+            assertEquals(TaskOutcome.SUCCESS, task(":lintKotlin")?.outcome)
+            assertEquals(expectedEmptyPlain(), reportContent("/main-lint.txt"))
+            assertEquals(expectedEmptyCheckstyle(), reportContent("main-lint.xml"))
+            assertEquals(expectedEmptyHtml(), reportContent("/main-lint.html"))
+            assertEquals(expectedEmptyJson(), reportContent("/main-lint.json"))
+            assertTrue(reportContent("/main-lint.sarif.json").isNotBlank())
```

Comment:
`sarif` report contains absolute paths which varies between devices. I haven't found any better way to ensure correct report gets generated

Reply:
Yeah, that is kind of annoying that Sarif doesn't keep to the convention of the others, but nonetheless.